.github/workflows/publish-docs.yml

```diff
@@ -1,28 +1,52 @@
 name: Publish Docs in GitHub Pages
+
 on:
+  workflow_dispatch: # allow manual runs
   push:
     branches:
       - main
+    paths: # avoid extra builds
+      - docs/**
+      - mkdocs.yml
+      - .github/workflows/publish-docs.yml
```

Comment:
FYI I'm not completely sure it's necessary to explicitly mention the action itself in the paths, but I'm including it for extra visibility into what triggers it, even if it may be superfluous.